src/librustc_mir/interpret/terminator.rs

```diff
@@ -463,7 +451,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
     /// Evaluate a const function where all arguments (if any) are zero-sized types.
     /// The evaluation is memoized thanks to the query system.
-    fn eval_const_fn_call(
+    pub (crate) fn eval_const_fn_call(
```

Comment:
Does this function need to live inside the Miri engine at all? Looks like most everything it does, `const_eval.rs` could also do.

Reply:
Though I feel like that file could really benefit from being split into the basic CTFE machine definition, and the rest... (similar to `machine.rs` vs `eval.rs` in Miri).

Reply:
I'm going to open a separate PR for pulling out all CTFE code to where it is actually needed

Reply:
*All* CTFE code? Isn't eval_const_fn_call the only one that's inside interpret/ when it shouldn't be?
I cleaned this up during my refactorings, and I don't think anything besides this fn here sneaked in since then. But I might have missed some PRs?

Reply:
Well I was gonna do a full review again, maybe that's the only one

Reply:
Okay. Then please just add a FIXME in this PR.